common/str-base.cpp

```diff
@@ -37,6 +37,7 @@ MemoryPool *g_refCountPool = nullptr; // FIXME: This is never freed right now
 Mutex *g_refCountPoolMutex = nullptr;
 
 void lockMemoryPoolMutex() {
+#ifndef SINGLE_THREADED
```

Comment:
Is it necessary to modify here as well?
I understand that speed matters but is this change really significant?
It would be best to not scatter such ifdef around all the code.

Reply:
It surely is.

Before:   `12.20%                   168364805                       Common::unlockMemoryPoolMutex()`

After:    `7.41%                    63310898                       Common::unlockMemoryPoolMutex()`

So it's about 40% faster (the number of % means % of CPU cycles spent in that function against the overall number of cycles spent during runtime).